### PR TITLE
work on capston project

### PR DESCRIPTION
**Pull request by hasnatali1947**
- completed my capstone project
- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Used API
- Use store, actions, and reducers in React.
- Use styles in a React app.

### CODE REVIEW

**Review by aamirkhan2478 — COMMENTED** [review]
Hi @hasnatali1947 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: :red_circle: :red_circle: 

You have submitted a project, that hasn't a descriptive PR summary kindly add a descriptive PR summary and resubmit it check below :point_down:  the example of a descriptive PR summary :slightly_smiling_face: 

Screenshot from 2023-05-26 20-12-14


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1 +1,123 @@
...
+
+### Tech Stack <a name="tech-stack"></a>
+
+HTML | CSS | JS | ES6 | React
+
+- More details about the project can be seen in this [video](https://www.loom.com/share/18778594bc524f93bfeabd8d35a33ca3) 
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [live link](https://hasnatali1947.github.io/Weather-App/)
```
Comment (t-yanick) [inline README.md]: - [x] Great job working on the react capstone project. As per proeject requirements, you are supposed to deploy your app and add the live link to your README. 
**Live version**
```
- The student deploys the application using Render, Netlify or GitHub Pages.
- The project is deployed properly, the home and detail page works as expected.
- A link to the deployed project is present in the README.
```
Unfortunately, when I try to visit your webpage from the shared [link](https://hasnatali1947.github.io/Weather-App/), I get the the page as per on the screenshot below. Kindly fix this deployment issue.
image

File: README.md [hunk 2]
```diff
@@ -1 +1,123 @@
...
+
+If you like this project please leave a ⭐️
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+Thanks Microverse, learn how to code > [Join Microverse](https://www.microverse.org/?grsf=9m3hq6)
```
Comment (t-yanick) [inline README.md]: - [x] As per the project requiements, you are expected to give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/) under the **Acknowledgements** section. Kindly do this.

File: README.md [hunk 3]
```diff
@@ -1 +1,123 @@
...
+
+- **[Search your city]**
+- **[Go to detail page]**
+- **[React Router Dom]**
+
+### Tech Stack <a name="tech-stack"></a>
+
+HTML | CSS | JS | ES6 | React
+
+- More details about the project can be seen in this [video](https://www.loom.com/share/18778594bc524f93bfeabd8d35a33ca3)
```
Comment (t-yanick) [inline README.md]: - [ ] Please kindly check your video presentation again. I am unable to play it and I cannot get the Transcript as well through which I could follow your presentation. Thanks

File: src/components/home.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  return (
+    <>
+      <div className="form-container">
+        <form onSubmit={handleSubmit}>
+          <input
+            className="searchcity"
+            value={country}
+            onChange={(e) => setCountry(e.target.value)}
+            placeholder="Search your city"
```
Comment (t-yanick) [inline src/components/home.js:67]: - [x] When I try to search a City, there's no filtering that occurs. For example, I wanted to search London, but when I type in the search box, all the cities are still displayed. See screenshot below:
image
Kindly fix your filtering to provide just the data being searched.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @hasnatali1947 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark: Project contains Home and Detail pages as per project requirements
:heavy_check_mark: Project well structured
:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly delete all the files generated when you create a new react project. This will make your folder structure clean and professional.
- [ ] Use Filter for stateless components to filter data retrieved from the API. You have not used any filter, probably that is the reason, search by city is not working. Kindly review this aspect of the project.
- [ ] Add unit tests for several components of the app and some pure functions in Redux code. Your app is currently not having enough tests.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/home.js (hunk 4 above)
Comment (hasnatali1947) [reply src/components/home.js:67]: It works by searching the name
image

**Hit Enter or click on the search icon**

The city would be added
image


Also, details are working properly
image

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @hasnatali1947 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights

- [ ] ✔️ Project contains Home and Detail pages as per project requirements
- [ ] ✔️ Project well structured
- [ ] ✔️ Good commit messages
- [ ] ✔️ Descriptive Readme file

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As suggested previously Kindly delete all the files generated when you create a new react project. This will make your folder structure clean and professional.
- [ ] As suggested previously Add unit tests for several components of the app and some pure functions in the Redux code. Your app currently not having enough tests
- [ ] As suggested previously Use Filter for stateless components to filter data retrieved from the API. You have not used any filter, probably that is the reason, the search by city is not working. Kindly review this aspect of the project.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hasnatali1947** [comment]
- [x] When I try to search a City, there's no filtering that occurs. For example, I wanted to search London, but when I type in the search box, all the cities are still displayed. See screenshot below:
image
Kindly fix your filtering to provide just the data being searched.

It works by searching the name
image

**Hit Enter or click on the search icon**

The city would be added
image


Also, details are working properly
image

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @hasnatali1947, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**